ql/src/java/org/apache/hadoop/hive/ql/parse/ColumnStatsSemanticAnalyzer.java

```diff
@@ -673,8 +672,8 @@ public ASTNode rewriteAST(ASTNode ast, ColumnStatsAutoGatherContext context)
     checkForPartitionColumns(colNames,
         Utilities.getColumnNamesFromFieldSchema(tbl.getPartitionKeys()));
     validateSpecifiedColumnNames(colNames);
-    if (conf.getBoolVar(ConfVars.HIVE_STATS_COLLECT_PART_LEVEL_STATS) && tbl.isPartitioned() 
-          && !tbl.hasNonNativePartitionSupport()) {
+    if (conf.getBoolVar(ConfVars.HIVE_STATS_COLLECT_PART_LEVEL_STATS) && tbl.isPartitioned()
```

Comment:
this condition is used at least twice, does it make sense to put into a separate method?

Reply:
extracted to StatUtils